hydra-http.c

```diff
@@ -248,6 +259,38 @@ int32_t start_http(int32_t s, char *ip, int32_t port, unsigned char options, cha
   ptr = ((char *)strchr(http_buf, ' '));
   if (ptr != NULL)
     ptr++;
+
+  // check status
+  if (match_status_code != NULL && match_status_code[0] != 0) {
```

Comment:
match_status_code can never be NULL. switch this to:
`if (match_status_code_cnt) {`